cli/server_test.go

```diff
@@ -1208,17 +1208,15 @@ func TestServer(t *testing.T) {
 				}
 			}
 			return htmlFirstServedFound
-		}, testutil.WaitMedium, testutil.IntervalFast, "no html_first_served telemetry item")
+		}, testutil.WaitLong, testutil.IntervalSlow, "no html_first_served telemetry item")
 	})
 	t.Run("Prometheus", func(t *testing.T) {
 		t.Parallel()
 
 		t.Run("DBMetricsDisabled", func(t *testing.T) {
 			t.Parallel()
 
-			ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitShort)
-			defer cancel()
-
+			ctx := testutil.Context(t, testutil.WaitLong)
 			randPort := testutil.RandomPort(t)
```

Comment:
In my experience, this is a huge culprit for this test flaking. But fixing it is tricky, we'd have to 1) start a listener and hand it over to the cmd, or 2) use `:0` but somehow signal the test what port to access.

Reply:
Yeah agreed -- follow-up